packages/api/src/controllers/helpers.ts

```diff
@@ -299,12 +299,47 @@ export async function sendgridEmail({
 export async function sendgridEmailPaymentFailed({
   email,
   sendgridApiKey,
-  userId,
+  user,
   invoiceId,
+  invoiceUrl,
+  templateId,
 }) {
+  // If user.lastPaymentFailed is newer than 3 days, don't send email
+  if (
+    user.lastPaymentFailed &&
+    user.lastPaymentFailed > Date.now() - 3 * 24 * 60 * 60 * 1000
+  ) {
+    return false;
+  }
```

Comment:
**nit:** Can you move this check to whoever calls this function? This function has a single purpose of sending an email, also reflected on its name, and I think it's misleading to start adding other logic to it like awareness about the whole building strategy.

Just for code clarity though, not a required change.